src/instructlab/data/generate.py

```diff
@@ -34,11 +34,18 @@
     default=DEFAULTS.CHUNK_WORD_COUNT,
     show_default=True,
 )
+# TODO - DEPRECATED - Remove in a future release
 @click.option(
     "--num-instructions",
     type=click.INT,
-    help="Number of instructions to generate.",
-    default=DEFAULTS.NUM_INSTRUCTIONS,
+    default=-1,
+    hidden=True,
+)
+@click.option(
+    "--sdg-scale-factor",
+    type=click.INT,
+    help="Number of instructions to generate for each seed example. The examples map to sample q&a pairs for new skills. For knowledge, examples are generated with both the sample q&a pairs, as well as chunks of the knowledge document(s), so the resulting data set is typically larger for a knowledge addition for the same value of `--sdg-scale-factor`.",
+    default=DEFAULTS.SDG_SCALE_FACTOR,
```

Comment:
The default should not be necessary. Click should look up the value from the config.

In fact you should be able to drop all defaults from click options that have a key in the configuration system. PR #1622 will help you to debug this behavior.

Reply:
perfect, yep, removed this line and the default still works properly